File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-3.java
class Solution {
    public int removeDuplicates(int[] nums) {
        if(nums.length == 1) return 1;

        int min = nums[0];
        int removedCount = 0;
        for(int i = 1; i < nums.length; i++){
            if (nums[i] == min){
                nums[i] = -101;
                removedCount++;
           } else {
            min = nums[i];
           }
        }

        System.out.println(Arrays.toString(nums));

        for(int i = 1; i < nums.length; i++){
           if(nums[i] == -101) {
                for(int j = i +1; j < nums.length; j++){
                    if(nums[j] != -101) {
                        nums[i] = nums[j];
                        nums[j] = -101;
                        break;
                    }
                }
           }
        }
       
        return nums.length - removedCount;
    }
}

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-3.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        if len(nums) == 1:
            return nums[0]

        mid = len(nums) // 2 
        left = self.findMin(nums[:mid])
        right = self.findMin(nums[mid:])

        return min(left, right)
        